contracts/Organization.sol

```diff
@@ -0,0 +1,272 @@
+ */
+contract Organization {
+
+    /* Usings */
+
+    using SafeMath for uint256;
+
+
+    /* Events */
+
+    event OwnershipTransferInitiated(address indexed _proposedOwner);
+
+    event OwnershipTransferCompleted(
+        address indexed _previousOwner,
+        address indexed _newOwner
+    );
+
+    event AdminAddressChanged(address indexed _newAdmin);
+
+    event WorkerSet(
+        address indexed _worker,
+        uint256 _expirationHeight,
+        uint256 _remainingHeight
+    );
+
+    event WorkerRemoved(
+        address indexed _worker
+    );
+
+
+    /* Storage */
+
+    /** Address for which private key will be owned by organization. */
+    address public owner;
+
+    /** proposedOwner is address proposed by owner for ownership transfer. */
+    address public proposedOwner;
+
+    /** admin address set by owner to facilitate operations of an economy. */
+    address public admin;
+
+    /**
+     *  List of whitelisted workers active upto the expiration height.
+     *  Expiration height 0 is a valid height.
+     */
+    mapping(address => uint256 /* expiration height */) public workers;
+
+
+    /* Modifiers */
+
+    modifier onlyOwner()
+    {
+        require(
+            msg.sender == owner,
+            "Only owner is allowed to call."
+        );
+        _;
+    }
+
+    modifier onlyOrganization()
+    {
+        require(
+            (msg.sender == owner) || (msg.sender == admin),
+            "Only owner/admin is allowed to call."
+        );
+        _;
+    }
+
+
+    /* Special Functions */
+
+    constructor() public
+    {
+        owner = msg.sender;
+    }
+
+
+    /* External Functions */
+
+    /**
+     * @notice Initiates ownership transfer to proposed owner.
+     *
+     * @dev Requires:
+     *          - msg.sender should be owner.
+     *      Allows resetting of owner to 0x address.
+     *
+     * @param _proposedOwner worker address to be added.
+     *
+     * @return true on successful execution.
+     */
+    function initiateOwnershipTransfer(address _proposedOwner)
+        external
+        onlyOwner
+        returns (bool)
+    {
+        require(
+            _proposedOwner != owner,
+            "ProposedOwner address can't be current owner address."
+        );
+
+        proposedOwner = _proposedOwner;
+
+        emit OwnershipTransferInitiated(_proposedOwner);
+
+        return true;
+    }
+
+    /**
+     * @notice Complete ownership transfer to proposed owner.
+     *
+     * @dev Requires:
+     *          - msg.sender should be proposed owner.
+     *
+     * @return true on successful execution.
+     */
+    function completeOwnershipTransfer() external returns (bool)
+    {
+        require(
+            msg.sender == proposedOwner,
+            "Caller is not proposed owner address."
+        );
+
+        address oldOwner = owner;
+        owner = proposedOwner;
+        proposedOwner = address(0);
+
+        emit OwnershipTransferCompleted(oldOwner, owner);
```

Comment:
:) Yes I discussed with team members(internal) and they don't any issue with that. I am listing my arguments below:
- oldOwner makes sure old owner is present in blockchain as datapoint. You can go to last event and know what was the previous owner.
- Tried to find ownership transfer related EIPs. Found one(draft phase) which replicates the event signature. https://eips.ethereum.org/EIPS/eip-173
- Doesn't cost much gas.

Reply:
"Don't see any issue" =/= see a value

I see your arguments, but my counter-arguments remain:
- the additional information provided may be useful, but is there any evidence to suggest not having the additional information has been at all harmful?
- while that EIP does support your proposal, it is still a draft, which almost be definition means we should not make changes based on it
- that is doesn't cost much gas is an acknowledgement that it still costs more gas so what I would need is a reason why we should add to the gas bill—that we are only adding a little is not a strong reason for adding any

There is also, presumably, the additional cost of having to change any consuming javascript work (e.g., I presume there are openst deployment services that are looking for events with 1 parameter, not 2, and that those will need to be updated).

As noted above, I will accept this if it is insisted upon (because, at the end of the day, it is a small change with limited knock-on effects), but I earnestly believe that your arguments, which are fair and reasonable, are still outweighed by my counter-arguments and, consequently, I remain unconvinced that this change is worth making.

Reply:
Just to update here:
I presume there are openst deployment services that are looking for events with 1 parameter, not 2, and that those will need to be updated

_OpenST.js hasn't integrated still with Organization contract._

Reply:
I assumed that integration would simply be copy/paste of the analogous deployment code; still, I take your point that the consuming JS does not technically yet exist—that point in my argument can fall away.